src/EnergyPlus/ExtendedHeatIndex.cc

```diff
@@ -0,0 +1,552 @@
+        constexpr Real64 eta = 1.43e-6;    // kg/J      , "inhaled mass" / "metabolic rate", steadman1979
+        Real64 L = Le(310.);               //           , latent heat of vaporization at 310 K
+        constexpr Real64 p = 1.013e5;      // Pa        , atmospheric pressure
+        constexpr Real64 rgasa = 287.04;   // J/kg/K
+        constexpr Real64 rgasv = 461.;     // J/kg/K
+        constexpr Real64 cva = 719.;       // J/kg/K
+        constexpr Real64 cvv = 1418.;      // J/kg/K
+        constexpr Real64 cpa = cva + rgasa;
+        constexpr Real64 cpv = cvv + rgasv;
+
+        return eta * Q * (cpa * (Tc - Ta) + L * rgasa / (p * rgasv) * (Pc - Pa));
+    }
+
+    // Function to calculate mass transfer resistance through skin, Pa m^2/W
+    Real64 Zs(Real64 const Rs)
+    {
+        return (Rs == 0.0387) ? 52.1 : 6.0e8 * std::pow(Rs, 5);
+    }
+
+    // Function to calculate heat transfer resistance through air, exposed part of skin, K m^2/W
+    Real64 Ra(Real64 const Ts, Real64 const Ta)
+    {
+        constexpr Real64 sigma = 5.67e-8; // W/m^2/K^4 , Stefan-Boltzmann constant
+        constexpr Real64 epsilon = 0.97;  //           , emissivity of surface, steadman1979
+        constexpr Real64 hc = 17.4;
+        constexpr Real64 phi_rad = 0.85;
+        Real64 const hr = epsilon * phi_rad * sigma * (std::pow(Ts, 2) + std::pow(Ta, 2)) * (Ts + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    // Function to calculate heat transfer resistance through air, clothed part of skin, K m^2/W
+    Real64 Ra_bar(Real64 const Tf, Real64 const Ta)
+    {
+        constexpr Real64 sigma = 5.67e-8; // W/m^2/K^4 , Stefan-Boltzmann constant
+        constexpr Real64 epsilon = 0.97;  //           , emissivity of surface, steadman1979
+        constexpr Real64 hc = 11.6;
+        constexpr Real64 phi_rad = 0.79;
+        Real64 const hr = epsilon * phi_rad * sigma * (std::pow(Tf, 2) + std::pow(Ta, 2)) * (Tf + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    // Function to calculate heat transfer resistance through air, when being naked, K m^2/W
+    Real64 Ra_un(Real64 const Ts, Real64 const Ta)
+    {
+        constexpr Real64 sigma = 5.67e-8; // W/m^2/K^4 , Stefan-Boltzmann constant
+        constexpr Real64 epsilon = 0.97;  //           , emissivity of surface, steadman1979
+        constexpr Real64 hc = 12.3;
+        constexpr Real64 phi_rad = 0.80;
+        Real64 const hr = epsilon * phi_rad * sigma * (std::pow(Ts, 2) + std::pow(Ta, 2)) * (Ts + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    constexpr Real64 Za = 60.6 / 17.4;     // Pa m^2/W, mass transfer resistance through air, exposed part of skin
+    constexpr Real64 Za_bar = 60.6 / 11.6; // Pa m^2/W, mass transfer resistance through air, clothed part of skin
+    constexpr Real64 Za_un = 60.6 / 12.3;  // Pa m^2/W, mass transfer resistance through air, when being naked
+
+    constexpr Real64 tol = 1e-8;
```

Comment:
This is an absolute tolerance level. What are the reasonable ranges of values here and what level of precision does this tolerance correspond to relative to those values?

Reply:
What someone else chose for the tolerance is not relevant. What is the magnitude of the values produced by this function? 1e-4? 1e-1? 1e+2?

Reply:
If this function produces SGP (whatever that is) and the values are on the order of 1e-2 then a tolerance of 1e-8 is about six decimal places of precision which is about twice what is needed. I cannot conceive of any quantity that EnergyPlus that would need to know to a precision of more than 0.1% (1% may be okay for many thing too).  

In fact, `SolveRoot` should probably be revised to use relative tolerance after some small fixed number of iterations. I wonder how much performance over-specific root precision is costing us. It could be quite a lot.

Reply:
Absolutely. Those additional digits are costing you iterations for no real benefit in terms of accuracy. Who cares if the SGP is "off" by less than 0.1%? How accurate is that metric anyway?

Reply:
Normally heat index is reported at either integer or 1 decimal point. The range is mostly two three hundred. Hopefully within 0.1C accuracy is good enough.

Reply:
Most thermostats used in buildings do not have this resolution.

Reply:
yeah the inputs of temperature and relative humidity percentage points are integers